modules/motions/ui/MotionFormStartNew/Parts/StartNewTopUpWithLimitsAndCustomToken.tsx

```diff
@@ -0,0 +1,333 @@
+              ))}
+            </SelectControl>
+          </Fieldset>
+          {selectedTokenAddress && (
+            <MotionInfoBox>
+              <Text as="span" size={12} weight={500}>
+                {selectedTokenLabel || 'Token'} address:{' '}
+              </Text>
+              <AddressInlineWithPop
+                address={selectedTokenAddress}
+                trim={false}
+              />
+            </MotionInfoBox>
+          )}
+          {periodLimitsData?.periodData && selectedTokenAddress && (
+            <MotionLimitProgressWrapper>
+              <MotionLimitProgress
+                spentAmount={periodLimitsData.periodData.alreadySpentAmount}
+                totalLimit={periodLimitsData.limits.limit}
+                startDate={periodLimitsData.periodData.periodStartTimestamp}
+                endDate={periodLimitsData.periodData.periodEndTimestamp}
+                token={selectedTokenLabel}
+                newAmount={newAmount}
+              />
+            </MotionLimitProgressWrapper>
+          )}
+
+          {fieldsArr.fields.map((item, fieldIdx) => (
+            <Fragment key={item.id}>
+              <FieldsWrapper>
+                <FieldsHeader>
+                  <FieldsHeaderDesc>Recipient #{fieldIdx + 1}</FieldsHeaderDesc>
+                  {fieldsArr.fields.length > 1 && (
+                    <RemoveItemButton
+                      onClick={() => fieldsArr.remove(fieldIdx)}
+                    >
+                      Remove recipient {fieldIdx + 1}
+                    </RemoveItemButton>
+                  )}
+                </FieldsHeader>
+                {periodLimitsData?.isEndInNextPeriod && (
+                  <MotionInfoBox>
+                    The motion is ending in the next period. The transfer limit
+                    would be replenished by that time.
+                  </MotionInfoBox>
+                )}
+                <Fieldset>
+                  <SelectControl
+                    label="Top up recipient address"
+                    name={`${fieldNames.programs}.${fieldIdx}.address`}
+                    rules={{ required: 'Field is required' }}
+                  >
+                    {getFilteredOptions(fieldIdx).map((program, j) => (
+                      <Option
+                        key={j}
+                        value={program.address}
+                        children={program.title}
+                      />
+                    ))}
+                  </SelectControl>
+                </Fieldset>
+
+                <Fieldset>
+                  <InputNumberControl
+                    label={`${selectedTokenLabel} Amount`}
+                    name={`${fieldNames.programs}.${fieldIdx}.amount`}
+                    disabled={!selectedTokenAddress}
+                    rules={{
+                      required: 'Field is required',
+                      validate: value => {
+                        const check1 = validateToken(value)
```

Comment:
could you give verbose name for `check1`

Reply:
Done!